client/doc.go

```diff
@@ -0,0 +1,156 @@
+
+One crucial component of the CONIKS key management system are the clients
+verifying the cryptographic proofs returned by the CONIKS server as part of
+the registration, lookup and monitoring protocols. client implements the
+operations performed by CONIKS clients during these protocols. This document
+outlines each of these protocols from a client perspective.
+
+For the server-side specification of the CONIKS protocols, see
+https://godoc.org/github.com/coniks-sys/coniks-go/server
+
+Registration
+
+Prerequisites: This protocol requires that the client pin both the
+server's public signing key for verifying signed tree roots (STRs) and
+the server's initial private index verification key (i.e. the public
+VRF key). We assume a separate PKI exists to manage the server's keys.
+
+Use of a CONIKS account verification bot is optional (see
+https://godoc.org/github.com/coniks-sys/coniks-go/bots for details).
+If the corresponding CONIKS key server requires that all registered names
+be validated by a CONIKS registration bot, the client also pins the
+registration bot that proxies all of its registration requests.
+Additionally, as the network of CONIKS auditors is currently very small,
+the client also maintains a list of pinned auditors. In a future release,
+we plan to introduce an automatic update mechanism for the auditors list
+while the development of a more scalable auditor discovery mechanism
+is underway.
+
+Protocol:
+- The user enters her username into the CONIKS client GUI prompt.
+
+- The client generates a new public-private key pair for this username,
+and stores the keys in secure? local storage on the device.
+
+- The client sends a registration request
+        reg_req = (username, key)
+to the server, or to the account validation bot, if used. See
+https://godoc.org/github.com/coniks-sys/coniks-go/bots for the
+CONIKS account validation protocol specification.
+
+- If the registration request is accepted*, the server returns a
+registration proof.
+More specifically, the server returns a proof of absence:
+        reg_pf = (auth_path, str)
+If the server uses the temporary binding protocol extension, reg_pf will
+also include a temporary binding (TB). See
+https://godoc.org/github.com/coniks-sys/coniks-go/protocol/extensions/tb
+for the temporary binding protocol extension specification.
+
+- The client audits the received STR including a hash chain check.
+See https://godoc.org/github.com/coniks-sys/coniks-auditor for the
+auditing protocol specification. *Note: Auditing is not implemented yet.*
+
+- Assuming the audit passes, the client proceeds to verify the TB, if
+included: it verifies the server's signature on th TB and checks that
+they public key in the TB matches the public key it sent to the proxy
+in reg_req.
+
+- Next, the client verifies that auth_path is a proof of absence by
+checking that its private index is invalid for the username, and its
+commitment is invalid for the public key it registered in reg_req.
```

Comment:
This is my mistake. I thought our verification of a proof of absence also does a commitment verification.